src/wallet/wallet.h

```diff
@@ -993,6 +993,7 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
     bool NewKeyPool();
     size_t KeypoolCountExternalKeys() EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
     bool TopUpKeyPool(unsigned int kpSize = 0);
+    void AddKeypoolPubkey(const CPubKey& pubkey, const bool internal);
```

Comment:
Nit: call this new function from `TopUpKeyPool`, so it's more clear this function reuses existing code. It could take a `error_message` and an optional `WalletBatch` argument. If `WalletBatch` is set it should skip `NotifyCanGetAddressesChanged()`, though that gets a bit ugly.

Reply:
I added a function `AddKeypoolPubkeyWithDB` which `TopUpKeypool` calls. `AddKeypoolPubkey` will make the `WalletBatch`, call `AddKeypoolPubkeyWithDB` and then ``NotifyCanGetAddressesChanged``.